context/src/main/java/io/grpc/Deadline.java

```diff
@@ -231,6 +232,30 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
+  @Override
+  public int hashCode() {
+    return Arrays.asList(this.ticker, this.deadlineNanos).hashCode();
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof Deadline)) {
+      return false;
+    }
+
+    final Deadline other = (Deadline) o;
+    if (this.ticker == null ? other.ticker != null : !this.ticker.equals(other.ticker)) {
```

Comment:
`ticker` should be compared by reference identity to be consistent with `compareTo()`.